pkg/controller/utils/clusterinfo_test.go

```diff
@@ -0,0 +1,201 @@
+package utils
+
+import (
+	"fmt"
+	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cloud-ingress-operator/pkg/testutils"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestClusterBaseDomain(t *testing.T) {
+	infraObj := testutils.CreateInfraObject("basename", testutils.DefaultAPIEndpoint, testutils.DefaultAPIEndpoint, testutils.DefaultRegionName)
+	objs := []runtime.Object{infraObj}
+	mocks := testutils.NewTestMock(t, objs)
+
+	base, err := GetClusterBaseDomain(mocks.FakeKubeClient)
+	if err != nil {
+		t.Fatalf("Could not get cluster base domain name: %v", err)
+	}
+	if base != "unit.test" {
+		t.Fatalf("Base domain mismatch. Expected %s, got %s", "unit.test", base)
+	}
+}
+
+func TestGetClusterPlatform(t *testing.T) {
+	infraObj := testutils.CreateInfraObject("basename", testutils.DefaultAPIEndpoint, testutils.DefaultAPIEndpoint, testutils.DefaultRegionName)
+	objs := []runtime.Object{infraObj}
+	mocks := testutils.NewTestMock(t, objs)
+
+	platform, err := GetClusterPlatform(mocks.FakeKubeClient)
+	if err != nil {
+		t.Fatalf("Couldn't get cluster platform: %v", err)
+	}
+	if platform != string(configv1.AWSPlatformType) {
+		t.Fatalf("Expected platform to be %v, got %v instead.", string(configv1.AWSPlatformType), platform)
+	}
+}
+
+func TestGetClusterName(t *testing.T) {
+	clustername := "cluster-test-name"
+	infraObj := testutils.CreateInfraObject(clustername, testutils.DefaultAPIEndpoint, testutils.DefaultAPIEndpoint, testutils.DefaultRegionName)
+	objs := []runtime.Object{infraObj}
+	mocks := testutils.NewTestMock(t, objs)
+
+	name, err := GetClusterName(mocks.FakeKubeClient)
+	if err != nil {
+		t.Fatalf("Couldn't get cluster name %v", err)
+	}
+	if name != clustername {
+		t.Fatalf("Expected cluster name to be %s, got %s instead", clustername, name)
+	}
+}
+
+func TestGetMasterSubnetNames(t *testing.T) {
+	clustername := "subnet-test"
+	masterNames := make([]string, 3)
+	for i := 0; i < 3; i++ {
+		masterNames[i] = fmt.Sprintf("master-%d", i)
+	}
+
+	machineList, _ := testutils.CreateMachineObjectList(masterNames, "ids", "master", testutils.DefaultRegionName, testutils.DefaultAzName)
+	infraObj := testutils.CreateInfraObject(clustername, testutils.DefaultAPIEndpoint, testutils.DefaultAPIEndpoint, testutils.DefaultRegionName)
+	objs := []runtime.Object{infraObj, machineList}
+	mocks := testutils.NewTestMock(t, objs)
+	subnetmap, err := GetMasterNodeSubnets(mocks.FakeKubeClient)
+
+	if err != nil {
+		t.Fatalf("Couldn't get subnet map: %v", err)
+	}
+	if len(subnetmap) != 2 {
+		t.Fatalf("Expected Subnet map to have exactly two keys, but got %d keys back", len(subnetmap))
+	}
+	if _, ok := subnetmap["public"]; !ok {
```

Comment:
The `_` (or whatever value there) is scoped to the `if` block. So to reuse later, something like this would be needed:

```go
var publicSubnet string
if publicSubnet, ok = subnetmap["public"]; !ok {
 t.Fatalf("failure")
}

if publicSubnet != expected {
  t.Fatalf("failure")
}
```

My opinion here is that semantically it isn't an improvement because the `publicSubnet` value is not useful anywhere; referring to it as `subnetmap["public"]` is fine.